contracts/cw1-whitelist/src/contract.rs

```diff
@@ -1,243 +1,451 @@
 use crate::error::ContractError;
-use cosmwasm_std::{Addr, Deps, DepsMut, Empty, Env, MessageInfo, Order, Response};
+use crate::responses::AdminListResponse;
+use cosmwasm_std::{Addr, Deps, DepsMut, Empty, Env, MessageInfo, Order, Response, StdResult};
 
-use cw1::{Cw1, FindMemberResponse};
-use cw_storage_plus::Map;
-use schemars::JsonSchema;
-use serde::{Deserialize, Serialize};
+use cw2::set_contract_version;
+use cw_storage_plus::{Item, Map};
 use sylvia::contract;
 
-#[derive(Serialize, Deserialize, Clone, PartialEq, Eq, JsonSchema, Debug)]
-pub struct MembersCntResponse {
-    pub cnt: usize,
-}
+const CONTRACT_NAME: &str = env!("CARGO_PKG_NAME");
+const CONTRACT_VERSION: &str = env!("CARGO_PKG_VERSION");
 
-pub struct Cw1WhitelistContract {
-    members: Map<'static, Addr, Empty>,
+pub struct Cw1WhitelistContract<'a> {
+    pub(crate) admins: Map<'static, &'a Addr, Empty>,
+    pub(crate) mutable: Item<'static, bool>,
 }
 
-impl Cw1 for Cw1WhitelistContract {
-    type Error = ContractError;
-    fn add_member(
-        &self,
-        (deps, _env, _info): (
-            cosmwasm_std::DepsMut,
-            cosmwasm_std::Env,
-            cosmwasm_std::MessageInfo,
-        ),
-        member: String,
-    ) -> Result<Response, Self::Error> {
-        self.members
-            .save(deps.storage, deps.api.addr_validate(&member)?, &Empty {})?;
-
-        Ok(Response::new())
-    }
-
-    fn find_member(
-        &self,
-        (deps, _env): (cosmwasm_std::Deps, cosmwasm_std::Env),
-        member: String,
-    ) -> Result<FindMemberResponse, Self::Error> {
-        let is_present = self
-            .members
-            .has(deps.storage, deps.api.addr_validate(&member)?);
-
-        Ok(FindMemberResponse { is_present })
-    }
-}
-
-#[contract(error=ContractError)]
+#[contract]
 #[messages(cw1 as Cw1)]
-impl Cw1WhitelistContract {
+impl Cw1WhitelistContract<'_> {
     pub const fn new() -> Self {
         Self {
-            members: Map::new("members"),
+            admins: Map::new("admins"),
+            mutable: Item::new("mutable"),
         }
     }
 
     #[msg(instantiate)]
     pub fn instantiate(
         &self,
-        (deps, _env, _msg): (DepsMut, Env, MessageInfo),
-        members: Vec<String>,
+        ctx: (DepsMut, Env, MessageInfo),
```

Comment:
Maybe just `(deps, _, _): (DepsMut, Env, MessageInfo),`

I know it's not the preatiest approach, but you do it later in line 34 anyway.
Until we will introduce #30 I think we can keep it as it was with only the difference to change `_env` to just `_`